File: OC.UI/src/app/blci/api/branches/ministry.ts
import { Pledge } from "../accounting/pledge";
import { BranchMinistry } from "./branch-ministry";


export interface Ministry {
    id?: number;
    name?: string;
    shortName?: string | null;
    description?: string | null;
    isActive?: boolean;
    branchMinistries?: BranchMinistry[];
    pledges?: Pledge[];
}

File: OC.Domain/Models/Branches/Ministry.cs
using OC.Domain.Interfaces;
using OC.Domain.Models.Accounting;

namespace OC.Domain.Models.Branches
{
    public class Ministry : AuditableEntity, IEntityBase
    {
        public Ministry() {
            BranchMinistries = new HashSet<BranchMinistry>();
            Pledges= new HashSet<Pledge>();
        }

        public long Id { get; set; }
        public string Name { get; set; }
        public string? ShortName { get; set; }
        public string? Description { get; set; }

        // Relationships
        public virtual ICollection<BranchMinistry> BranchMinistries { get; set; }
        public virtual ICollection<Pledge> Pledges { get; set; }

    }
}
